src/Sparrow/Logging/LoggingSource.cs

```diff
@@ -108,7 +108,14 @@ public async Task Register(WebSocket source, WebSocketContext context, Cancellat
             {
                 if (_listeners.TryAdd(source, context) == false)
                     throw new InvalidOperationException("Socket was already added?");
-                SetupLogMode(LogMode, _path, RetentionTime, RetentionSize, Compressing);
```

Comment:
The cause of the test failure is that the `RemoveSocket` creates a new log thread async and it can happen after the test is finished and after `EndLogging` is called. The solution can be just to be sure in the test that `EndLogging` is called after the new log thread is set but using the same thread seem to be better implementation (and it fixes the test too).
Is there a benefit of creating a new thread?